File: frontend/src/api/folderNoteAPI.js
import { http } from './http.js';

// Folder API
export async function getFolders() {
  return http('/folders');
}

export async function createFolder(folderName, parentFolderID = null) {
  return http('/folders', { method: 'POST', body: { folderName, parentFolderID } });
}

export async function renameFolder(folderID, folderName) {
  return http(`/folders/${folderID}`, { method: 'PUT', body: { folderName } });
}

export async function deleteFolder(folderID) {
  return http(`/folders/${folderID}`, { method: 'DELETE' });
}

export async function assignDocumentToFolder(documentID, folderID) {
  return http('/folders/assign', { method: 'POST', body: { documentID, folderID } });
}

export async function unassignDocumentFromFolder(documentID) {
  return http('/folders/unassign', { method: 'POST', body: { documentID } });
}

// Note API
export async function getNotes(documentID) {
  return http(`/notes/document/${documentID}`);
}

export async function createNote(documentID, noteTitle, noteContent) {
  return http('/notes', { method: 'POST', body: { documentID, noteTitle, noteContent } });
}

export async function updateNote(noteID, noteTitle, noteContent) {
  return http(`/notes/${noteID}`, { method: 'PUT', body: { noteTitle, noteContent } });
}

export async function deleteNote(noteID) {
  return http(`/notes/${noteID}`, { method: 'DELETE' });
}

export async function getNoteCounts() {
  return http('/notes/counts');
}

File: docmanage_mobile/lib/api/folder_note_api.dart
import 'http_service.dart';

class FolderNoteApi {
  // =========================================================
  // FOLDERS
  // =========================================================

  static Future getFolders() async {
    return HttpService.request('/folders', method: 'GET');
  }

  static Future createFolder(String folderName, {int? parentFolderID}) async {
    return HttpService.request('/folders', method: 'POST', body: {
      'folderName': folderName,
      if (parentFolderID != null) 'parentFolderID': parentFolderID,
    });
  }

  static Future renameFolder(int folderID, String folderName) async {
    return HttpService.request('/folders/$folderID', method: 'PUT', body: {
      'folderName': folderName,
    });
  }

  static Future deleteFolder(int folderID) async {
    return HttpService.request('/folders/$folderID', method: 'DELETE');
  }

  static Future assignDocument(int documentID, int folderID) async {
    return HttpService.request('/folders/assign', method: 'POST', body: {
      'documentID': documentID,
      'folderID': folderID,
    });
  }

  static Future unassignDocument(int documentID) async {
    return HttpService.request('/folders/unassign', method: 'POST', body: {
      'documentID': documentID,
    });
  }

  // =========================================================
  // NOTES
  // =========================================================

  static Future getNotes(int documentID) async {
    return HttpService.request('/notes/document/$documentID', method: 'GET');
  }

  static Future createNote(int documentID, String? title, String content) async {
    return HttpService.request('/notes', method: 'POST', body: {
      'documentID': documentID,
      'noteTitle': title,
      'noteContent': content,
    });
  }

  static Future updateNote(int noteID, String? title, String content) async {
    return HttpService.request('/notes/$noteID', method: 'PUT', body: {
      'noteTitle': title,
      'noteContent': content,
    });
  }

  static Future deleteNote(int noteID) async {
    return HttpService.request('/notes/$noteID', method: 'DELETE');
  }

  static Future getNoteCounts() async {
    return HttpService.request('/notes/counts', method: 'GET');
  }
}
